store/src/java/com/zimbra/cs/account/Provisioning.java

```diff
@@ -1310,7 +1310,8 @@ public SetPasswordResult setPassword(Account acct, String newPassword, boolean e
         throw ServiceException.UNSUPPORTED();
     }
 
-    public abstract void resetPassword(Account acct, String newPassword, boolean dryRun) throws ServiceException;
+    public abstract void verifyDryRun(Account acct, String newPassword, boolean enforcePasswordPolicy,
```

Comment:
Any reason we can not use prov.checkPasswordStrength? 
Adding abstract method will break certain implementations. 
Like you have fixes some but you never know if any of the server extensions or client customization that we are not aware of.

Reply:
please find a better way for password history.

Reply:
Was using it for verifying password history also, as per your suggestion implemented it separately and used `checkPasswordStrength` method.

Reply:
**Adding abstract method will break certain implementations.**  
I am asking to avoid adding abstract method.